libcommon/proto.c

```diff
@@ -82,14 +82,32 @@ uint8_t readbyte()
 	}
 }
 
-int read(uint8_t *buf, size_t bufsize, size_t nbytes)
+uint8_t readbyte(uint8_t *mode, uint8_t *mode_bytes_left)
+{
+	if (*mode_bytes_left == 0) {
+		*mode = readbyte_();
+		if ((*mode == MODE_CDC) || (*mode == MODE_HID) ||
+		    (*mode == MODE_TKEYCTRL)) {
+			*mode_bytes_left = readbyte_();
+		} else {
+			debug_puts("We only support MODE_CDC, MODE_HID and "
+				   "MODE_TKEYCTRL\n");
```

Comment:
What happens here after debug_puts()? Will continue directly, right? Perhaps an assert()?

Reply:
That's correct. I've added an assert.